TestsFoundation/TestsFoundation/Extensions/XCUIApplicationExtensions.swift

```diff
@@ -17,34 +17,36 @@
 import Foundation
 import XCTest
 
+// warning: These functions will actually use `app` for finding elements,
+// so will only function properly on the test target app.
 public extension XCUIApplication {
     func find(label: String, type: XCUIElement.ElementType = .any) -> Element {
        return XCUIElementWrapper(
-            descendants(matching: type)
+            app.descendants(matching: type)
```

Comment:
I'm not sure I follow the reasoning here. Isn't `app` recomputing `XCUIApplication()`? So calling `app.find(label:)` does a compute twice?

Reply:
before it was an implicit `self.`. Because it's now an autoclosure that gets called each time we do something with the element we needed an explicit `self.`. However we don't want to re-use self each time, since that would be referencing a potentially stale object. `app` gets us a new one each time, but it's not guaranteed that the original app referenced the same bundle.

Reply:
oh, yes. `app.find()` will do another app compute inside it. that's probably unnecessary.